### PR TITLE
integ-tests: FSx Storage Capacity/File Chunk Size/Maintenance Start Time

### PR DESCRIPTION
**Pull request by yuleiwan**
* Ensure FSx File Chunk Size/Maintenance Start Time retrieved from cluster are same as config file setting
* Ensure deviation of FSx storage capacity is less than 0.2 between the storage size in cluster and config file

Signed-off-by: Yulei Wang <yuleiwan@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 1]
```diff
@@ -520,6 +542,50 @@ def _test_data_repository_task(remote_command_executor, mount_dir, bucket_name,
     assert_that(file_permissions).is_equal_to("0100777")
 
 
+def _test_storage_capacity(remote_command_executor, mount_dir, storage_capacity):
+    logging.info("Test FSx storage capacity")
+    if storage_capacity:
+        result = remote_command_executor.run_remote_command(
+            f"df -BG | grep {mount_dir}" + "| awk '{print$2}' | tr -d -c 0-9"
+        )
+        assert_that((storage_capacity - int(result.stdout)) // storage_capacity).is_less_than(0.2)
```
Comment (rexcsn) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:551]: Might be worth adding some explanation in comments whenever we have hardcoded tolerances like this. Any documentation explaining this? Will this tolerance be the same for all FSx storage type?

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 2]
```diff
@@ -144,6 +160,9 @@ def _test_fsx_lustre(
     storage_type=None,
     auto_import_policy=None,
     deployment_type=None,
+    weekly_maintenance_start_time=None,
+    imported_file_chunk_size=None,
+    storage_capacity=None,
```
Comment (rexcsn) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:165]: This `_test_fsx_lustre` function is getting very complicated and these default parameters are getting a bit out of hand here :sweat_smile:

IMO we should extract the common assertions also used by other test into a separate function like `_assert_basic_fsx_functions` and rename this function to something like `_assert_fsx_configuration_options` and get rid of all the default `=None`

The issue I see with `=None` here is we don't really need them here, and having them here will not produce an obvious failure if we decreased test coverage, because no error will be shown if we remove any of these parameters by accident

This way you can also get rid of all the conditions within your assertion functions like `if storage_capacity:`

The bottom line is we should always be enforcing our assertions strictly in our tests. If things change, we would want to see an error rather than not being aware of the issue.

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 3]
```diff
@@ -520,6 +542,50 @@ def _test_data_repository_task(remote_command_executor, mount_dir, bucket_name,
     assert_that(file_permissions).is_equal_to("0100777")
 
 
+def _test_storage_capacity(remote_command_executor, mount_dir, storage_capacity):
+    logging.info("Test FSx storage capacity")
+    if storage_capacity:
```
Comment (rexcsn) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:547]: Can we remove these? These conditions will fail silently if we decreased coverage by accident

Please double check and change for all other assertions as well

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 4]
```diff
@@ -520,6 +542,50 @@ def _test_data_repository_task(remote_command_executor, mount_dir, bucket_name,
...
+def _test_weekly_maintenance_start_time(fsx_fs_id, weekly_maintenance_start_time, region):
+    logging.info("Test FSx weekly maintenance start time setting")
+    if weekly_maintenance_start_time:
+        assert_that(get_weekly_maintenance_start_time(fsx_fs_id, region)).is_equal_to(weekly_maintenance_start_time)
+
+
+def get_imported_chunch_size(fsx_fs_id, region):
+    logging.info("Getting ImportedFileChunkSize from DescribeFilesystem API.")
+    return (
+        boto3.client("fsx", region_name=region)
```
Comment (rexcsn) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:569]: Can we just describe the file system once and parse different info out of the result in different tests?

Please double check for other assertions/test cases that may need to describe the file system as well

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 5]
```diff
@@ -157,6 +176,9 @@ def _test_fsx_lustre(
     _test_export_path(remote_command_executor, mount_dir, bucket_name, region)
     _test_auto_import(auto_import_policy, remote_command_executor, mount_dir, bucket_name, region)
     _test_data_repository_task(remote_command_executor, mount_dir, bucket_name, fsx_fs_id, region)
+    _test_storage_capacity(remote_command_executor, mount_dir, storage_capacity)
```
Comment (rexcsn) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:179]: We will want to include `_test_storage_capacity` for our tests that are specifying, but currently not testing storage capacity, like `test_fsx_lustre`.

Please double check for all existing tests

File: tests/integration-tests/tests/storage/test_fsx_lustre.py (hunk 2 above)
Comment (yuleiwan) [reply tests/integration-tests/tests/storage/test_fsx_lustre.py]: Done

File: tests/integration-tests/tests/storage/test_fsx_lustre.py (hunk 3 above)
Comment (yuleiwan) [reply tests/integration-tests/tests/storage/test_fsx_lustre.py]: Done

File: tests/integration-tests/tests/storage/test_fsx_lustre.py (hunk 4 above)
Comment (yuleiwan) [reply tests/integration-tests/tests/storage/test_fsx_lustre.py]: Done

File: tests/integration-tests/tests/storage/test_fsx_lustre.py (hunk 5 above)
Comment (yuleiwan) [reply tests/integration-tests/tests/storage/test_fsx_lustre.py]: Done

File: tests/integration-tests/tests/storage/test_fsx_lustre.py (hunk 1 above)
Comment (yuleiwan) [reply tests/integration-tests/tests/storage/test_fsx_lustre.py]: add a comment to explain it

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 6]
```diff
@@ -85,13 +88,58 @@ def test_fsx_lustre_configuration_options(
...
+    storage_capacity,
+):
+    remote_command_executor = RemoteCommandExecutor(cluster)
+    scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
+    fsx_fs_id = get_fsx_fs_id(cluster, region)
+    fsx = boto3.client("fsx", region_name=region).describe_file_systems(FileSystemIds=[fsx_fs_id])
+
+    _test_fsx_lustre_correctly_mounted(remote_command_executor, mount_dir, os, region, fsx_fs_id)
+    _test_import_path(remote_command_executor, mount_dir)
+    _test_fsx_lustre_correctly_shared(scheduler_commands, remote_command_executor, mount_dir)
```
Comment (rexcsn) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:132]: Minor: instead of rewriting these common test cases can we just call `_test_fsx_lustre`?

Comment (yuleiwan) [reply tests/integration-tests/tests/storage/test_fsx_lustre.py]: Done

File: tests/integration-tests/tests/storage/test_fsx_lustre.py [hunk 7]
```diff
@@ -520,6 +550,41 @@ def _test_data_repository_task(remote_command_executor, mount_dir, bucket_name,
...
 
 
+def _test_storage_capacity(remote_command_executor, mount_dir, storage_capacity):
+    logging.info("Test FSx storage capacity")
+    # Get the storage size of mount_dir with GB, the storage size is slightly less than storage capacity defined,
+    # here we set 0.2 as the difference ratio threshold
+    result = remote_command_executor.run_remote_command(
+        f"df -BG | grep {mount_dir}" + "| awk '{print$2}' | tr -d -c 0-9"
+    )
+    assert_that((storage_capacity - int(result.stdout)) // storage_capacity).is_less_than(0.2)
```
Comment (rexcsn) [inline tests/integration-tests/tests/storage/test_fsx_lustre.py:560]: Incorrect use of `//` here? `1//2 = 0`

Comment (yuleiwan) [reply tests/integration-tests/tests/storage/test_fsx_lustre.py]: Done

**Review by rexcsn — APPROVED** [review]
LGTM! Feel free to merge after successful test run.
